[Site Isolation] Add some tests covering iframe sandboxing

achristensen07 opened the pull request:
#### 51b8fa6d46a537290eefc24d9702a8be5dea4297
<pre>
[Site Isolation] Add some tests covering iframe sandboxing
<a href="https://bugs.webkit.org/show_bug.cgi?id=279479">https://bugs.webkit.org/show_bug.cgi?id=279479</a>
<a href="https://rdar.apple.com/135761965">rdar://135761965</a>

Reviewed by Charlie Wolfe.

I started implementing this for site isolation in a separate PR,
<a href="https://github.com/WebKit/WebKit/pull/33423">https://github.com/WebKit/WebKit/pull/33423</a>
but I decided the test infrastructure was substantial enough for its own PR.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm:
(TestWebKitAPI::siteIsolatedViewAndDelegate):
(TestWebKitAPI::TEST(SiteIsolation, PreferencesUpdatesToAllProcesses)):
(TestWebKitAPI::TEST(SiteIsolation, ParentOpener)):
(TestWebKitAPI::TEST(SiteIsolation, ChildBeingNavigatedToMainFrameDomainByParent)):
(TestWebKitAPI::TEST(SiteIsolation, ChildBeingNavigatedToSameDomainByParent)):
(TestWebKitAPI::TEST(SiteIsolation, RunOpenPanel)):
(TestWebKitAPI::TEST(SiteIsolation, AppKitText)):
(TestWebKitAPI::TEST(SiteIsolation, SetFocusedFrame)):
(TestWebKitAPI::TEST(SiteIsolation, EvaluateJavaScriptInFrame)):
(TestWebKitAPI::TEST(SiteIsolation, MainFrameURLAfterFragmentNavigation)):
(TestWebKitAPI::TEST(SiteIsolation, CustomUserAgent)):
(TestWebKitAPI::TEST(SiteIsolation, ApplicationNameForUserAgent)):
(TestWebKitAPI::TEST(SiteIsolation, WebsitePoliciesCustomUserAgent)):
(TestWebKitAPI::TEST(SiteIsolation, WebsitePoliciesCustomNavigatorPlatform)):
(TestWebKitAPI::TEST(SiteIsolation, ProvisionalLoadFailureOnCrossSiteRedirect)):
(TestWebKitAPI::TEST(SiteIsolation, SynchronouslyExecuteEditCommandSelectAll)):
(TestWebKitAPI::TEST(SiteIsolation, SelectAll)):
(TestWebKitAPI::TEST(SiteIsolation, CanGoBackAfterLoadingAndNavigatingFrame)):
(TestWebKitAPI::TEST(SiteIsolation, CanGoBackAfterNavigatingFrameCrossOrigin)):
(TestWebKitAPI::TEST(SiteIsolation, NavigateIframeSameOriginBackForward)):
(TestWebKitAPI::TEST(SiteIsolation, NavigateIframeCrossOriginBackForward)):
(TestWebKitAPI::TEST(SiteIsolation, NavigateNestedIframeSameOriginBackForward)):
(TestWebKitAPI::TEST(SiteIsolation, AdvancedPrivacyProtectionsHideScreenMetricsFromBindings)):
(TestWebKitAPI::TEST(SiteIsolation, UpdateWebpagePreferences)):
(TestWebKitAPI::TEST(SiteIsolation, SandboxFlags)):
* Tools/TestWebKitAPI/cocoa/TestWKWebView.h:
* Tools/TestWebKitAPI/cocoa/TestWKWebView.mm:
(-[TestWKWebView firstChildFrame]):

Canonical link: <a href="https://commits.webkit.org/283481@main">https://commits.webkit.org/283481@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/81aa4f6eba5cac14f243286829b78992a3a5d2ff

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/66378 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/45753 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/18999 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/70411 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/16989 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/68496 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/53552 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/17271 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/53236 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/11849 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/69445 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/42175 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/57462 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/33890 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/38846 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/14850 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/15864 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/60741 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/15192 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/72114 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/10335 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/14564 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/60559 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/10367 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/57531 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/60873 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/8527 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/2143 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10061 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/41560 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/42637 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/43820 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/42380 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

charliewolfe commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm:
```diff
@@ -3173,4 +3169,134 @@ HTTPServer server({
+    EXPECT_FALSE(receivedOpen);
+
+    reset();
+    [webView evaluateJavaScript:@"i.src = 'https://example.org/check-when-loaded'" completionHandler:nil];
```
Did you mean for this to be `example.org`? You're using `example.com` above.

achristensen07 replied:
yeah, I wanted them to be distinct domains so site isolation will put them in different processes.

achristensen07 committed:
[Site Isolation] Add some tests covering iframe sandboxing
https://bugs.webkit.org/show_bug.cgi?id=279479
rdar://135761965

Reviewed by Charlie Wolfe.

I started implementing this for site isolation in a separate PR,
https://github.com/WebKit/WebKit/pull/33423
but I decided the test infrastructure was substantial enough for its own PR.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm:
(TestWebKitAPI::siteIsolatedViewAndDelegate):
(TestWebKitAPI::TEST(SiteIsolation, PreferencesUpdatesToAllProcesses)):
(TestWebKitAPI::TEST(SiteIsolation, ParentOpener)):
(TestWebKitAPI::TEST(SiteIsolation, ChildBeingNavigatedToMainFrameDomainByParent)):
(TestWebKitAPI::TEST(SiteIsolation, ChildBeingNavigatedToSameDomainByParent)):
(TestWebKitAPI::TEST(SiteIsolation, RunOpenPanel)):
(TestWebKitAPI::TEST(SiteIsolation, AppKitText)):
(TestWebKitAPI::TEST(SiteIsolation, SetFocusedFrame)):
(TestWebKitAPI::TEST(SiteIsolation, EvaluateJavaScriptInFrame)):
(TestWebKitAPI::TEST(SiteIsolation, MainFrameURLAfterFragmentNavigation)):
(TestWebKitAPI::TEST(SiteIsolation, CustomUserAgent)):
(TestWebKitAPI::TEST(SiteIsolation, ApplicationNameForUserAgent)):
(TestWebKitAPI::TEST(SiteIsolation, WebsitePoliciesCustomUserAgent)):
(TestWebKitAPI::TEST(SiteIsolation, WebsitePoliciesCustomNavigatorPlatform)):
(TestWebKitAPI::TEST(SiteIsolation, ProvisionalLoadFailureOnCrossSiteRedirect)):
(TestWebKitAPI::TEST(SiteIsolation, SynchronouslyExecuteEditCommandSelectAll)):
(TestWebKitAPI::TEST(SiteIsolation, SelectAll)):
(TestWebKitAPI::TEST(SiteIsolation, CanGoBackAfterLoadingAndNavigatingFrame)):
(TestWebKitAPI::TEST(SiteIsolation, CanGoBackAfterNavigatingFrameCrossOrigin)):
(TestWebKitAPI::TEST(SiteIsolation, NavigateIframeSameOriginBackForward)):
(TestWebKitAPI::TEST(SiteIsolation, NavigateIframeCrossOriginBackForward)):
(TestWebKitAPI::TEST(SiteIsolation, NavigateNestedIframeSameOriginBackForward)):
(TestWebKitAPI::TEST(SiteIsolation, AdvancedPrivacyProtectionsHideScreenMetricsFromBindings)):
(TestWebKitAPI::TEST(SiteIsolation, UpdateWebpagePreferences)):
(TestWebKitAPI::TEST(SiteIsolation, SandboxFlags)):
* Tools/TestWebKitAPI/cocoa/TestWKWebView.h:
* Tools/TestWebKitAPI/cocoa/TestWKWebView.mm:
(-[TestWKWebView firstChildFrame]):

Canonical link: https://commits.webkit.org/283481@main
